token-service/src/main/java/org/p2p/token/service/model/TokenServiceResult.kt

```diff
@@ -3,12 +3,12 @@ package org.p2p.token.service.model
 sealed class TokenServiceResult<T> {
     data class Success<T>(val data: T) : TokenServiceResult<T>()
 
-    sealed class Error(val cause: Throwable) : TokenServiceResult<Throwable>()
+    data class Error<T>(val cause: Throwable) : TokenServiceResult<T>()
 }
 
 fun <T> TokenServiceResult<T>.unwrap(): T? {
     return when (this) {
         is TokenServiceResult.Success<T> -> this.data
-        is TokenServiceResult.Error -> throw cause
+        is TokenServiceResult.Error<T> -> null
```

Comment:
no need to change unwrap contract
you can create a new method if you need to 

.successOrNull()

unwrap is a convention naming in case of error inside this error would be thrown